### ISSUE TITLE
Style/RedundantRegexpCharacterClass false positive

### ISSUE THREAD

**Issue by marcandre** [rubocop-hq/rubocop#8593]
```ruby
/#{Regexp.union(
  %w"( ) { } [ ] < > $ ! ^ ` ... + * ? ,"
)}/o
```

is incorrectly detected and corrected...

```
[Corrected] Style/RedundantRegexpCharacterClass: Redundant single-element character class, [ ] can be replaced with  .
  %w"( ) { } [ ] < > $ ! ^ ` ... + * ? ,"
             ^^^
```

This cop should use the Regexp parser and only check the literal sections.
